pkg/operatorsource/updater.go

```diff
@@ -0,0 +1,113 @@
+package operatorsource
+
+import (
+	"context"
+	"errors"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+	"github.com/operator-framework/operator-marketplace/pkg/datastore"
+	"github.com/operator-framework/operator-marketplace/pkg/phase"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// RegistryUpdater is an interface that can be used to check whether a remote registry
+// has any update(s) and trigger a rebuild of in cluster OperatorSource cache.
+type RegistryUpdater interface {
```

Comment:
FYI, the filename is still `updater.go`. Shouldn't that be `registry_updater.go`?